repo/repo.go

```diff
@@ -100,6 +109,40 @@ func ImportRepoRules(filename string, repoCache *RemoteCache) ([]*rule.Rule, err
 	return rules, nil
 }
 
+// FilterRulesByFile filters a list of generated repo rules by the file where
+// the rule should be written. This is done by comparing the generated
+// rules with the current list of repo rules declared in the repository. If
+// the generated rule matches a rule already in the repo, then it inherits the
+// file where the rule was defined. Else its file is set as the destFile parameter.
+func FilterRulesByFile(before []Repo, after []*rule.Rule, destFile *rule.File) map[*rule.File][]*rule.Rule {
```

Comment:
`before` and `after` are a little misleading: it makes it sound like `after` is an out parameter that will be written into. `repos` and `genRules` would be fine. These are the names the caller uses.

Also, this function should not be exported. It's only needed within this package.

Reply:
`FilterRulesByFile` is called outside of this package by `update-repos.go`. Are you saying that the code should be refactored in a way that the function does not need to be exported?

Reply:
Oops, you're right.

Let's replace this with `func MergeRules(genRules []*rule.Rule, existingRules map[*rule.File][]string, destFile *rule.File) error` in the `repo` package. The second argument would be what `ListRepositories` would return. This would merge the generated rules into the appropriate files (or into `destFile` if they're new rules). With this function, we'd have common logic for writing the rules out.

This function probably belongs in the `merger` package, but I'm not confident enough about API stability to put it there. I consider `repo` less stable than `merger`, since most of the `repo` functionality should eventually be moved into `language/go`.